diesel_migrations/migrations_internals/src/migration.rs

```diff
@@ -49,6 +58,39 @@ pub fn migration_from(path: PathBuf) -> Result<Box<Migration>, MigrationError> {
     }
 }
 
+#[cfg(feature = "barrel")]
+struct BarrelMigration(PathBuf, String, String, String);
```

Comment:
Can we just move the migration trait from `diesel_migrations` into `diesel` so this code can live in Barrel?